### DISCUSSION TITLE
can you confirm version of zlint being used

### DISCUSSION THREAD

**Discussion by connellyt** [pkimetal/pkimetal#180] [Q&A] [answered]
I was doing some testing, and an attempted certificate issuance passed zlint validation with zlint alone, using version 3.6.4 released on October 12.  However, the same certificate failed via the zlint version in PKIMetal version 1.10.0.  

Looking at the changelog, it seems that pkimetal version 1.10.0 updated to zlint github.com/zmap/zlint/v3 v3.6.5-0.20241012161707-45a7d73dd7e2, but I don't see a 3.6.5 version on the zlint github page.

The specific lints failed, that I did not expect to fail, were:

STDOUT:     "Findings": [
STDOUT:         {
STDOUT:             "Linter": "certlint",
STDOUT:             "Finding": "BR certificates must include an HTTP URL of the OCSP responder",
STDOUT:             "Severity": "error"
STDOUT:         },
STDOUT:         {
STDOUT:             "Linter": "pkilint",
STDOUT:             "Finding": "A required element is absent",
STDOUT:             "Field": "certificate.tbsCertificate.extensions.1.extnValue.authorityInfoAccessSyntax",
STDOUT:             "Code": "cabf.serverauth.subscriber.ocsp_aia_access_method_absent",
STDOUT:             "Severity": "error"
STDOUT:         },
STDOUT:         {
STDOUT:             "Linter": "x509lint",
STDOUT:             "Finding": "No OCSP over HTTP",
STDOUT:             "Severity": "error"
STDOUT:         }
STDOUT:     ]
STDOUT: }

Can you confirm that the zlint in PKIMetal is the correct version?

Thanks,
Tom

**Comment by robstradling** [comment, marked as answer]
Hi @connellyt.  https://github.com/pkimetal/pkimetal/blob/v1.10.0/go.mod#L16 shows that the version of ZLint in pkimetal v1.10.0 is indeed v3.6.4.

The changelog entry you mentioned relates to https://github.com/pkimetal/pkimetal/blob/v1.10.0/dev_go.mod#L16, which is only used to build Development pkimetal releases.

ZLint v3.6.5 does not yet exist.  v3.6.5-0.20241012161707-45a7d73dd7e2 is a Go [pseudo-version number](https://go.dev/doc/modules/version-numbers#in-development).

> The specific lints failed, that I did not expect to fail, were:

None of those linter findings originated from ZLint.  One was from Certlint, one was from Pkilint, and one was from X509lint.
